doc/source/workflows/events.rst

```diff
@@ -99,6 +100,111 @@ After the workflow finishes checkpointing the event, the event listener will be
              self.consuemr.commit(event, asynchronous=False)
 
 
+
+Events via HTTP
+---------------
+
+An HTTP event is a specific workflow event received from an HTTP endpoint enabled
+by Ray serve. The implementation of the HTTP endpoint is named `HTTPEventProvider`. Its
+corresponding event listener is `HTTPListener`.
+
+Receive an event by HTTP
+--------------------------
+
+`HTTPListener` is a customized `EventListener` that subscribes to an `HTTPEventProvider`
+with the unique combination of `workflow_id` and `event_key`. An HTTP request from
+an external event submitter needs to specify `workflow_id` as part of the request URL and
+a JSON formatted document consisting of `event_key` and `event_payload`.
+
+The external submitter learns the values of `workflow_id` and `event_key` through
+the application context, which is out of the scope in this discussion.
+
+.. code-block:: python
+
+    import ray
+    from ray import workflow
+    from ray.workflow.http_event_provider import HTTPListener
+
+    # Create a task waiting for an http event with a JSON message.
+    # The JSON message is expected to have an event_key field
+    # and an event_payload field.
+    event_task = workflow.wait_for_event(HTTPListener, event_key='my_event_key')
+
+    key, messsage = workflow.run(event_task, workflow_id='workflow_receive_event_by_http')
+
+workflow_id as part of the request URL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, `HTTPEventProvider` is deployed as a Ray serve endpoint at
+`http://hostname:port/event/send_event/<workflow_id>`. If it is deployed locally,
```

Comment:
<img width="769" alt="image" src="https://user-images.githubusercontent.com/74173148/181853874-9e77c0e2-05c0-4d79-a6a6-dfe9e5cf2294.png">


It seems not correct

Reply:
Fixed